Backpack/Calendar/Classes/BPKCalendar.h

```diff
@@ -0,0 +1,75 @@
+/*
+ * Backpack - Skyscanner's Design System
+ *
+ * Copyright 2018 Skyscanner Ltd
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+#import "BPKCalendarAppearance.h"
+
+@class BPKCalendar;
+
+/**
+ * Enum values for specifying calendar selection type
+ */
+typedef NS_ENUM(NSUInteger, BPKCalendarSelection) {
+    BPKCalendarSelectionSingle = 0,
+    BPKCalendarSelectionRange = 1,
+    BPKCalendarSelectionMultiple = 2,
+};
+
+@protocol BPKCalendarDelegate
+
+/**
+ * Called when a date selection changed.
+ *
+ * @param calendar The backpack calendar.
+ * @param dateList List of selected dates.
+ */
+- (void)calendar:(BPKCalendar *)calendar didChangeDateSelection:(NSArray<NSDate *> *)dateList;
```

Comment:
How does this differentiated between range and single selection? Should we have a speific type for a range selection?

Reply:
In single selection mode the array will always have 0/1 element, in range selection mode two elements, in multi selection mode 0+ elements

Do we need to have a specific type? My first plan was to use NSDateInterval but this is only available from iOS10.0 onwards